### PR TITLE
spec/schema: change annotations to ACIdentifier

### PR DESCRIPTION
**Pull request by jonboulle**
https://github.com/appc/spec/commit/555e3e9b5e78781146eec8ff2cc30e5a27a0dafd
redefined ACName to a more restrictive syntax, and changed some fields
(like name, isolators, etc) to a new ACIdentifier type with lighter
restrictions similar to the previous ACName. Since annotations is a
field we expect to be namespaced it makes more sense for it to also use
the ACIdentifier type.

### CODE REVIEW

File: ace/validator.go [hunk 1]
```diff
@@ -350,7 +350,7 @@ func validateAppAnnotations(metadataURL string, pm *schema.PodManifest, app *sch
 			r = append(r, err)
 		}
 
-		lbl, err := types.NewACName(key)
+		lbl, err := types.NewACIdentifier(key)
 		if err != nil {
 			r = append(r, fmt.Errorf("invalid annotation label: %v", err))
```
Comment (yifan-gu) [inline ace/validator.go]: Should we replace `label` with `name` here to stay consistent with the doc?

File: schema/types/annotations_test.go [hunk 2]
```diff
@@ -199,8 +199,8 @@ func TestAnnotationsSet(t *testing.T) {
 
 	a.Set("dog", "woof")
 	w = Annotations{
-		Annotation{ACName("foo"), "bar"},
-		Annotation{ACName("dog"), "woof"},
+		Annotation{ACIdentifier("foo"), "bar"},
+		Annotation{ACIdentifier("dog"), "woof"},
```
Comment (yifan-gu) [inline schema/types/annotations_test.go]: maybe add some test cases such as `example.org/foo/foo_bar`

**Comment by jonboulle** [comment]
@yifan-gu updated

File: ace/validator.go [hunk 3]
```diff
@@ -287,7 +287,7 @@ func validatePodAnnotations(metadataURL string, pm *schema.PodManifest) results
 			r = append(r, err)
 		}
 
-		name, err := types.NewACName(key)
+		name, err := types.NewACIdentifier(key)
 		if err != nil {
 			r = append(r, fmt.Errorf("invalid annotation label: %v", err))
```
Comment (yifan-gu) [inline ace/validator.go]: here s/label/name/ as well?

Comment (jonboulle) [inline ace/validator.go]: fixed

**Comment by yifan-gu** [comment]
LGTM. Want another pair of eyes, but I think German team is :sleeping: this time...

**Comment by iaguis** [comment]
Noch nicht! LGTM

**Comment by jonboulle** [comment]
I would like to get another maintainer ack before landing.

File: schema/types/annotations_test.go (hunk 2 above)
Comment (philips) [inline schema/types/annotations_test.go]: +1

**Comment by philips** [comment]
lgtm, I agree with having one more test as @yifan-gu pointed out

**Comment by jonboulle** [comment]
I did add a case @ https://github.com/appc/spec/pull/442/files#diff-e0d5d10f9ae10b151ee373931cbaa0a3R214

**Pull request merged**